File: Java/1506-Find-Root-of-N-ary-Tree/soln.java
/*
// Definition for a Node.
class Node {
    public int val;
    public List<Node> children;

    
    public Node() {
        children = new ArrayList<Node>();
    }
    
    public Node(int _val) {
        val = _val;
        children = new ArrayList<Node>();
    }
    
    public Node(int _val,ArrayList<Node> _children) {
        val = _val;
        children = _children;
    }
};
*/

class Solution {
    public Node findRoot(List<Node> tree) {
        int val = 0;
        for(Node node : tree) {
            val += node.val;
            for(Node child : node.children) {
                val -= child.val;
            }
        }
        for(Node node : tree) {
            if(node.val == val) return node;
        }
        return null;
    }
}

File: C++/1506-Find-Root-of-N-ary-Tree/soln.cpp
/*
// Definition for a Node.
class Node {
public:
    int val;
    vector<Node*> children;

    Node() {}

    Node(int _val) {
        val = _val;
    }

    Node(int _val, vector<Node*> _children) {
        val = _val;
        children = _children;
    }
};
*/

class Solution {
public:
    Node* findRoot(vector<Node*> tree) {
        int val = 0;
        for(const auto & node : tree) {
            val += node->val;
            for(const auto & child : node->children) {
                val -= child->val;
            }
        }
        for(const auto & node : tree) {
            if(node->val == val) return node;
        }
        return nullptr;
    }
};